### PR TITLE
DOCK-2126: Fix weird github apps lambda behavior

### PR DESCRIPTION
**Pull request by svonworl**
This PR fixes the strange behavior that was reported in https://ucsc-cgl.atlassian.net/browse/DOCK-2126 dockstore/dockstore#4831.

TLDR: The lambda was not retrying requests when it received a 5xx from the webservice, causing pushes to be lost.  This PR fixes it.  And it's a negative-one-liner w00t!

A node.js lambda handler method is passed a callback, which is used by the lambda to return a result or signal an error condition:
https://docs.aws.amazon.com/lambda/latest/dg/nodejs-handler.html

After much investigation, I determined that the line that I removed was being executed before most other callback invocations, essentially preempting all subsequent retry decisions (most (all?) of which are made in event handlers).  Valid requests were only being retried in the relatively-rare case that the lambda itself threw an error (such as from the https lib if it couldn't connect to the server).

I confirmed this behavior with various experiments on the running dev lambda.

At first, I wanted to do a fairly extensive refactor, but I decided to keep things simple, not break anything, and focus the effort on more pressing matters.  We could add code to handle http network errors (`req.on("error")` ...), but I'm actually fine with the exception propagating outwards, and being handled by the queuing system (which correctly retries on exceptions).

Things I think I sorta might know about AWS lambdas, now:

1. The first callback invocation takes precedence over the rest.
2. Regarding retries, the queuing system only cares if the lambda error-ed, or not.  The statusCodes and messages that the lambda returns would be useful to the API Gateway service, but they don't matter to the queuing system.

I could be wrong about either, especially number 2.  AWS docs are not thorough and the hivemind does not offer clear, unanimous advice.

We should improve our lambda testing.  I'd add some, but this is my inaugural lambda PR, so I don't really have much insight about that, yet.  Given the scope, user testing it on dev should be fine...

### CODE REVIEW

**Comment by coverbeck** [comment]
I was going to say, PR should mention GitHub issue, but weirdly Unito is now doing something? I'd still mention the GitHub issue explicitly though.

File: upsertGitHubTag/deployment/index.js [hunk 1]
```diff
@@ -269,8 +269,6 @@ function processEvent(event, callback) {
         " from GitHub.",
     });
   }
-
-  callback(null, { statusCode: 200, body: "results" });
```
[the remark this answers is not in the document]
Comment (svonworl) [reply upsertGitHubTag/deployment/index.js:273]: Yeah, we probably should do something to prevent the subsequent code from being run.  The `else if` would be fine, but I'd be even more inclined to add a `return` at line 207.  That way we get the heck out of there, for sure.

**Review by denis-yuen — APPROVED** [review]
Neat!

> We should improve our lambda testing. I'd add some, but this is my inaugural lambda PR, so I 
don't really have much insight about that, yet. Given the scope, user testing it on dev should be fine...

Yes, this has been a bit of a pain and definitely worth a longer look at

**Review by fhembroff — APPROVED** [review]

File: upsertGitHubTag/deployment/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (wshands) [reply upsertGitHubTag/deployment/index.js:273]: Good catch, but why doesn't the handleCallback at line 234 execute first and send an error rather than line 273 first?

**Review by wshands — APPROVED** [review]

File: upsertGitHubTag/deployment/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (svonworl) [reply upsertGitHubTag/deployment/index.js:273]: In a nutshell, the `handleCallback` at line 235 (I added the `return` and bumped it down a line) is in the body of a callback function which is called in a function that is installed as a handler for the `end` event of the http request, 
https://github.com/dockstore/lambda/blob/2532890d5d372fffd99a84abaad9ce7ba6e0b763/upsertGitHubTag/deployment/index.js#L58
So, in that case, the lambda callback is invoked after the http response has been received.
